in-work/lln_clt.md

```diff
@@ -210,21 +214,175 @@ Moreover, if we repeat the exercise with a larger value of $n$, we should see th
+
+We can vary values for `n` to see how the distribution changes
 
 ```{code-cell} ipython3
-# TODO: write the code and put the plot here
+def generate_multiple_hist(X_distribution, ns, m, log_scale=False):
+    _, ax = plt.subplots(figsize=(10, 6))
+
+    def draw_means(X_distribution, n):
+        X_samples = X_distribution.rvs(size=n)
+        return np.mean(X_samples)
+   
+    for n in ns:
+        sample_means = [draw_means(X_distribution, n) for i in range(m)]
+        if log_scale:
+            plt.xscale('symlog')
+        ax.hist(sample_means, bins=40, alpha=0.4, density=True, label=fr'$n = {n}$')
+
+    mu = X_distribution.mean()
+    if not np.isnan(mu):
+        ax.axvline(x=mu, ls="--", lw=3, label=fr"$\mu = {mu}$")
+
+    ax.set_xlim(min(sample_means), max(sample_means)) 
+    ax.set_xlabel(r'$\bar x$', size=12)
+    ax.set_ylabel('density', size=12)
+    ax.legend()
+    plt.show()
 ```
 
+```{code-cell} ipython3
+generate_multiple_hist(st.norm(loc=5, scale=2), ns=[20_000, 50_000, 100_000], m=10_000)
+```
+
+We see that the histogram gradually converges to $\mu$.
+
+You can imagine the result when extrapolating this trend for $n \to \infty$.
+
+
 ## Breaking the LLN
 
 We have to pay attention to the assumptions in the statement of the LLN when we apply it.
 
-TODO
+As indicated by {eq}`lln_as`, LLN can break when $\mathbb E |X|$ is not finite or is not well defined.
+
+We can demonstrate this using a simple simulation using a [Cauchy distribution](https://en.wikipedia.org/wiki/Cauchy_distribution) for which it does not have a well-defined $\mu$.
+
+
+We lost the convergence we have seen before with normal distribution
+
+```{code-cell} ipython3
+fig, axes = plt.subplots(1, 2, figsize=(15, 6))
+
+def scattered_mean(distribution, burn_in, n, jump, ax, title, color, ylog=False):
+   
+  #Set a jump to reduce simulation complexity
+    sample_means = [np.mean(distribution.rvs(size=i)) 
+          for i in range(burn_in, n+1, jump)]
+   
+    ax.scatter(range(burn_in, n+1, jump), sample_means, s=10, c=color)
+   
+    #Change the y-axis to log scale if necessory
+    if ylog:
+        ax.set_yscale("symlog")
+    ax.set_title(title, size=10)
+    ax.set_xlabel(r"$n$", size=12)
+    ax.set_ylabel(r"$\bar x$", size=12)
+    yabs_max = max(ax.get_ylim(), key=abs)
+    ax.set_ylim(ymin=-yabs_max, ymax=yabs_max)
+    return ax
+
+scattered_mean(distribution=st.cauchy(), 
+             burn_in=1000, 
+             n=1_000_000, 
+             ax=axes[0],
+             jump=2000,
+             title="Cauchy Distribution",
+             color='#1f77b4',
+             ylog=True)
+
+scattered_mean(distribution=st.norm(), 
+             burn_in=1000, 
+             n=1_000_000,
+             ax=axes[1],
+             jump=2000,
+             title="Normal Distribution",
+             color='#ff7f0e')
+
+fig.suptitle('Sample Mean with Different Sample Sizes')
+plt.show()
```

Comment:
Maybe we can set the scales of the y-axis the **same** for the plots on the left and right of this panel so that it is easier to compare?

Alternatively, do you think it is a good idea to scatterplot these two distributions in the same figure but labelled with different colors?

Reply:
Many thanks for the suggestion @shlff. I think for this plot, the idea is to show the convergence and divergence, and the plots are on a very different scale (one in $\pm 10^2$, but the other is between $\pm 0.04$), so it would be hard to put them on the same scale. However, I find the axes are reversed for some reason. Thanks for helping me spotting this.